### DISCUSSION TITLE
Proposal for rn-reusables-cli

### DISCUSSION THREAD

**Discussion by eznix86** [founded-labs/react-native-reusables#16] [Ideas]
Right now, we have to clone the repository to get the components, instead we can have a cli similar to what the original project has.

I looked into how `shadcn-ui` cli works.

The way how it works is like this:
- generate the base configuration (in our cases install base utils and hooks needed later, such as global.css, nativewind etc)
- then pulling of components. (in our case, we can use github API to pull objects and create them locally)
Content of the registy:

https://ui.shadcn.com/registry

``` json
[
  {
    "name": "accordion",
    "dependencies": [
      "@radix-ui/react-accordion" // <---- package.json 
    ],
    "files": [
      "ui/accordion.tsx" // <---- location of the component
    ],
    "type": "components:ui"  <----type of the component
  },
  {
    "name": "alert",
    "files": [
      "ui/alert.tsx"
    ],
    "type": "components:ui"
  },
  {
    "name": "calendar",
    "dependencies": [ 
      "react-day-picker",
      "date-fns"
    ],
    "registryDependencies": [
      "button"  // <---- components dependencies
    ],
    "files": [
      "ui/calendar.tsx"
    ],
    "type": "components:ui"
  },
   ...
]
```
The type of the component, is necessary in our case because we will need some hooks etc. we can call it "lib:hooks", or something like this as components dependencies.

We can have a second repository to keep this json list. The original source code for the cli is found at https://github.com/shadcn-ui/ui/blob/main/packages/cli/src/index.ts

**Comment by mrzachnugent** [comment]
https://github.com/mrzachnugent/react-native-reusables/tree/main/packages/cli
